Sources/SlideButton/SlideButton.swift

```diff
@@ -22,17 +23,40 @@ public struct SlideButton: View {
     /// Use this initializer to create a new instance of `SlideButton` with the given title, styling, and callback. The `styling` parameter allows you to customize the appearance of the slide button, such as changing the size and color of the indicator, the alignment of the title text, and whether the text fades or hides behind the indicator. The `callback` parameter is executed when the user successfully swipes the indicator.
     ///
     /// - Parameters:
-    ///   - title: The title of the slide button.
     ///   - styling: The styling options to customize the appearance of the slide button. Default is `.default`.
     ///   - callback: The async callback that is executed when the user successfully swipes the indicator.
-    public init(_ title: String, styling: Styling = .default, callback: @escaping () async -> Void) {
-        self.title = title
+    ///   - label: The function creating a label view
+    
+    public init(styling: Styling = .default, callback: @escaping () async -> Void, @ViewBuilder label: () -> Label) {
+        self.title = label()
         self.callback = callback
         self.styling = styling
-        
+            
         self._offset = .init(initialValue: styling.indicatorSpacing)
     }
     
+    
+    @ViewBuilder
+    private var indicatorShape : some View {
+        switch styling.indicatorShape {
+        case .circular:
+            Circle()
+        case .rectangular(let cornerRadius):
+            RoundedRectangle(cornerRadius: cornerRadius ?? 0)
```

Comment:
This should be `RoundedRectangle(cornerRadius: max(0, (cornerRadius ?? 0) - styling.indicatorSpacing))` for an even spacing between the indicator and background